packages/camera/camera_platform_interface/lib/src/types/capture_mode.dart

```diff
@@ -0,0 +1,36 @@
+/// The mode the controller should operate in.
+///
+/// This capture mode determines whether the capture session is optimized for
+/// video recording or photo capture.
+///
+/// Defaults to [CaptureMode.video] as the camera plugin configuration is
+/// currently geared towards video recording.
+enum CaptureMode {
+  /// Capture a photo.
+  photo,
+
+  /// Capture a video, however this allows the user to take photos while recording.
+  video,
+}
+
+/// Returns the capture mode as a string
+String serializeCaptureMode(CaptureMode captureMode) {
+  switch (captureMode) {
+    case CaptureMode.photo:
+      return 'photo';
+    case CaptureMode.video:
+      return 'video';
+  }
+}
+
+/// Returns the capture mode for a given string.
+CaptureMode deserializeCaptureMode(String captureMode) {
```

Comment:
nit: You could also use the modern approach of using enhanced enums for this as well and add 

```dart
factory CaptureMode.deserialize(String captureMode) {
  switch(captureMode) {
    // ...
  }

  throw ArgumentError(`...`);
}
```

to `CaptureMode`.